plugins/hdfsviewer/src/azkaban/viewer/hdfs/velocity/hdfs-file-js.vm

```diff
@@ -130,7 +130,11 @@ azkaban.HdfsFileView = Backbone.View.extend({
     $('#file-contents-loading').hide();
 
     if (contentType == "HTML") {
-      // write to iframe
+      // Write content to iframe. We can't just set the html source content as
+      // inner html of the iframe. The iframe hosts a completely separate dom
+      // structure than the current html, thus setting the innher HTML will not
+      // work. The only way to store html inside iframe is by writing the content
```

Comment:
Typo? inner HTML?

What do you mean by "setting inner HTML" here? Could you elaborate more why it doesn't work? Any reference? 

Could you also explain why you have to use javascript to inject the content vs returning the content from server directly? 
Have you tried https://www.w3schools.com/tags/att_iframe_srcdoc.asp?

Reply:
Yeah that's a typo, will fix it. I will also update the doc with more detailed explanation.

The way Azkaban server works now is that it first generates a web skeleton (everything except the content), then sends an async request to fetch the content. So even if we use the srcdoc attribute, we have to inject the content into that attr using javascript.

Reply:
> The way Azkaban server works now is that it first generates a web skeleton (everything except the content), then sends an async request to fetch the content. So even if we use the srcdoc attribute, we have to inject the content into that attr using javascript.

Does that technique work? Have you tried?
You shouldn't be limited by what the existing AZ code does. It may be convenient to do. It doesn't mean it is the best way.
I saw that your justification for relaxing the sandbox is that it is needed somehow for the javascript method you used to work.

Reply:
You are right that I'm not limited to what the AZ framework does today, but to introduce a different behavior means there would be additional if/else logic in AZ code which reduces the elegance and readability of the code. 2nd, if we get both the web skeleton and html content together (and put html content in srcdoc like you said), for large html files, user would see a blank page for a few seconds before he could see the entire page, and they may think the server has dead while waiting. Following the current AZ framework is more user friendly since it shows the skeleton page and a progress bar to indicate data is being downloaded, so user know the server is responsive and nothing is broken. 3rd, it's not trivial to simply put content into the "srcdoc" attribute. If the html content is this: \<h3\>Hello " World\</h3\>, then putting it inside srcdoc will result in error: \<iframe srcdoc="\<h3\>Hello " World\</h3\>\</iframe\>. The quote in the h3 field will mess things up and allow sql injection. We can certainly escape such chars, but there's no guarantee that we can do it right or even cover everything we need. Overall I think fetching with an async call and write it into the iframe is the better option.